src/java.base/share/classes/sun/security/util/DisabledAlgorithmConstraints.java

```diff
@@ -808,6 +808,40 @@ private static class UsageConstraint extends Constraint {
         String[] usages;
         Set<SSLScope> scopes;
 
+        // To avoid user confusion, we block signature scheme names to be used
+        // with CertificateSignature algorithm constraints usage. For example,
+        // RSASSA-PSS certificate signature algorithm corresponds to multiple
+        // signature scheme names and blocking one of those signature scheme
+        // with CertificateSignature usage directive won't block RSASSA-PSS
+        // certificate signature because other rsa_pss_* signature schemes
+        // still will be allowed.
+        // Below are signature scheme names supported in TLSv1.2 and TLSv1.3
+        // that are unique (i.e. different from corresponding algorithm names).
+        // We avoid calling back the SSL layer to get these names because of
+        // the circular dependency.
+        private static final Set<String> BLOCKED_FOR_CERT_SCOPE = Set.of(
```

Comment:
Will we need to update the list whenever a new scheme is introduced? I've noticed all names contain an underscore and IIRC no standard name of a Signature has it. Do you think we can just check for the underscore character as a heuristic?